module-1/pandas-project/your-code/PandasProject.ipynb

```diff
@@ -1789,14 +1789,14 @@
    "cell_type": "markdown",
```

Comment:
1. Execution [3]: You don't need to write manually all the column labels, you can make a selection based on the columns function of the dataframe or you can define a list with the variables you want to use for dataframe slicing and then aply the iloc pandas function. The code would be more readable and pythonic way. 
2. Execution [7] : The same comment as in execution [3]
3. In real world, the assumption that the meaning of participant_relationship null values is "Relationship does not exist" should be double checked before filling null values. If you conclude that null values are collecting data errors, you would need to ignore this variable in your analysis. 
4. In[10] workingData.participant_relationship[workingData.participant_relationship!=0] = 1, I don't understand what to want to get with this and the code isn't commented. Take a look on the warnings jupyter kernel is throwing. 
5. In[14] Have you checked if the dataset has any 0 values? (0 are not null)
6. In[29] Very pro using loc function. Congrats!! Have you understood properly how it works? In case not, come back to me. 
7. 
"I have been trying to resolve a group by with several columns that aggregates a column and sort by it but I have been unable to achieve any result
dataWithBins.groupby(['bins', 'state']).count().sort_values("numberKilled")" . I need to explain it to you on your screen and with your code. Give me a hit when you have a free slot. 
8. In [41]: be careful, you have expanded your bins column to one column by bin. This can cause unexpected and not desired results in further steps of your analysis.